charmhelpers/contrib/openstack/context.py

```diff
@@ -746,7 +746,15 @@ def __call__(self):
 
         log('Generating template context for ceph', level=DEBUG)
         mon_hosts = []
+        # We set `key` and `auth` to None here, by default, to ensure
```

Comment:
Rather than set it to `None` here, why not override `context_complete` to check for the missing keys and not allow them to be `None` (which could change the meaning elsewhere)?

e.g.

```python
def context_complete(self, ctxt):
    """Override to ...
    
    :param ctxt: The current context members ...
    :type ctxt: Dict[str, ANY]
    :returns: True if the context is complete
    :rtype: bool
    """
    if 'auth' not in ctxt or 'key' not in ctxt:
        return False
    return super(CephContext, self).context_complete(ctxt)
```

Reply:
Sounds elegant :)